diff --git a/rtl/cv32e40x_alignment_buffer.sv b/rtl/cv32e40x_alignment_buffer.sv
index 2f252cea..99e89de2 100644
--- a/rtl/cv32e40x_alignment_buffer.sv
+++ b/rtl/cv32e40x_alignment_buffer.sv
@@ -147,8 +147,8 @@ module cv32e40x_alignment_buffer import cv32e40x_pkg::*;
   //////////////////
   // FIFO signals //
   //////////////////
-  inst_resp_t [0:ALBUF_DEPTH-1]  resp_q;
-  logic [0:ALBUF_DEPTH-1]        valid_n,   valid_int,   valid_q;
+  inst_resp_t [ALBUF_DEPTH-1:0]  resp_q;
+  logic [ALBUF_DEPTH-1:0]        valid_n,   valid_int,   valid_q;
   inst_resp_t resp_n;
 
   // Read/write pointer for FIFO
